src/platform/ESP32/OpenthreadLauncher.c

```diff
@@ -20,18 +20,82 @@
 #include "esp_netif.h"
 #include "esp_netif_types.h"
 #include "esp_openthread.h"
+#include "esp_openthread_cli.h"
 #include "esp_openthread_lock.h"
 #include "esp_openthread_netif_glue.h"
 #include "esp_openthread_types.h"
 #include "esp_vfs_eventfd.h"
 #include "freertos/FreeRTOS.h"
 #include "freertos/task.h"
+#include "openthread/cli.h"
 #include "openthread/instance.h"
 #include "openthread/logging.h"
 #include "openthread/tasklet.h"
 
 static esp_netif_t * openthread_netif                       = NULL;
 static esp_openthread_platform_config_t * s_platform_config = NULL;
+static TaskHandle_t cli_transmit_task                       = NULL;
+static QueueHandle_t cli_transmit_task_queue                = NULL;
+
+QueueHandle_t get_cli_transmit_task_queue(void)
+{
+    return cli_transmit_task_queue;
+}
+
+static int cli_output_callback(void * context, const char * format, va_list args)
+{
+    int ret = 0;
+    char prompt_check[3];
+    vsnprintf(prompt_check, sizeof(prompt_check), format, args);
+    if (!strncmp(prompt_check, "> ", sizeof(prompt_check)) && cli_transmit_task)
+    {
+        xTaskNotifyGive(cli_transmit_task);
+    }
+    else
+    {
+        ret = vprintf(format, args);
+    }
+    return ret;
+}
+
+static void esp_openthread_matter_cli_init(void)
+{
+    otCliInit(esp_openthread_get_instance(), cli_output_callback, NULL);
+}
+
+static void cli_transmit_worker(void * context)
+{
+    cli_transmit_task_queue = xQueueCreate(8, sizeof(char **));
+    if (!cli_transmit_task_queue)
+    {
+        return;
+    }
+    char * command_line = NULL;
+    while (true)
+    {
+        if (xQueueReceive(cli_transmit_task_queue, &command_line, portMAX_DELAY) == pdTRUE)
+        {
+            if (command_line)
+            {
+                esp_openthread_cli_input(command_line);
+                free(command_line);
+            }
+            else
+            {
+                continue;
+            }
+            xTaskNotifyWait(0, 0, NULL, portMAX_DELAY);
+        }
+    }
+    vQueueDelete(cli_transmit_task_queue);
+    vTaskDelete(NULL);
+}
+
+static esp_err_t cli_command_transmit_task(void)
+{
+    xTaskCreate(cli_transmit_worker, "ocli_trans", 3072, xTaskGetCurrentTaskHandle(), 4, &cli_transmit_task);
```

Comment:
should we add a constant for the priority 4? maybe with a comment on why that is guaranteed to run before anyone using cli_transmit_task_queue is trying to access the newly created queue

Reply:
To avoid the null cli_transmit_task_queue, function get_cli_transmit_task_queue is replaced by cli_transmit_task_post. The cli_transmit_task_queue is not operated directly by user.